modules/apps/object/object-api/src/main/java/com/liferay/object/related/models/NestedEntityObjectRelatedModelsProvider.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.object.related.models;
+
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.model.ObjectRelationship;
+
+/**
+ * @author Sergio Jiménez del Coso
+ */
+public interface NestedEntityRelatedModelsProvider {
+
+	public String getClassName();
+
+	public String getObjectRelationshipType();
+
+	public void validate(
+		Object propertyValue, ObjectEntry serviceBuilderObjectEntry,
```

Comment:
That is valid when we want to relate an ObjectEntry with another element, but is this be reusable for system objects?